homeassistant/components/axis/binary_sensor.py

```diff
@@ -23,26 +26,159 @@
 from .entity import AxisEventEntity
 from .hub import AxisHub
 
-DEVICE_CLASS = {
-    EventGroup.INPUT: BinarySensorDeviceClass.CONNECTIVITY,
-    EventGroup.LIGHT: BinarySensorDeviceClass.LIGHT,
-    EventGroup.MOTION: BinarySensorDeviceClass.MOTION,
-    EventGroup.SOUND: BinarySensorDeviceClass.SOUND,
-}
-
-EVENT_TOPICS = (
-    EventTopic.DAY_NIGHT_VISION,
-    EventTopic.FENCE_GUARD,
-    EventTopic.LOITERING_GUARD,
-    EventTopic.MOTION_DETECTION,
-    EventTopic.MOTION_DETECTION_3,
-    EventTopic.MOTION_DETECTION_4,
-    EventTopic.MOTION_GUARD,
-    EventTopic.OBJECT_ANALYTICS,
-    EventTopic.PIR,
-    EventTopic.PORT_INPUT,
-    EventTopic.PORT_SUPERVISED_INPUT,
-    EventTopic.SOUND_TRIGGER_LEVEL,
+
+@dataclass(frozen=True, kw_only=True)
+class AxisBinarySensorDescription(BinarySensorEntityDescription):
+    """Axis binary sensor entity description."""
+
+    event_topic: tuple[EventTopic, ...] | EventTopic
+    """Event topic that provides state updates."""
+    name_fn: Callable[[AxisHub, Event], str] = lambda hub, event: ""
+    """Function providing the corresponding name to the event ID."""
+    supported_fn: Callable[[AxisHub, Event], bool] = lambda hub, event: True
+    """Function validating if event is supported."""
+
+
+@callback
+def event_id_is_int(event_id: str) -> bool:
+    """Make sure event ID is int."""
+    try:
+        return isinstance(int(event_id), int)
+    except ValueError:
+        return False
+
+
+@callback
+def guard_suite_supported_fn(hub: AxisHub, event: Event) -> bool:
+    """Validate event ID is int."""
+    _, _, profile_id = event.id.partition("Profile")
+    return event_id_is_int(profile_id)
+
+
+@callback
+def object_analytics_supported_fn(hub: AxisHub, event: Event) -> bool:
+    """Validate event ID is int."""
+    _, _, profile_id = event.id.partition("Scenario")
+    return event_id_is_int(profile_id)
+
+
+@callback
+def guard_suite_name_fn(
+    handler: FenceGuardHandler
+    | LoiteringGuardHandler
+    | MotionGuardHandler
+    | Vmd4Handler,
+    event: Event,
+    event_type: str,
+) -> str:
+    """Get guard suite item name."""
+    if handler.initialized and (profiles := handler["0"].profiles):
+        for profile_id, profile in profiles.items():
+            camera_id = profile.camera
+            if event.id == f"Camera{camera_id}Profile{profile_id}":
+                return f"{event_type} {profile.name}"
```

Comment:
Shouldn't we make event type translatable?

Reply:
Those are names so not sure how applicable that is